homeassistant/components/evohome/__init__.py

```diff
@@ -221,9 +251,116 @@ def _handle_exception(err) -> bool:
         broker.update, config[DOMAIN][CONF_SCAN_INTERVAL]
     )
 
+    setup_service_functions(hass, broker)
+
     return True
 
 
+def setup_service_functions(hass: HomeAssistantType, broker):
+    """Set up the service functions.
+
+    Not all evohome-compatible systems support all system modes.
+    """
+
+    async def force_refresh(call):
+        """Obtain the latest state data via the vendor's RESTful API."""
+        await broker.update()
+
+    async def set_system_mode(call):
+        """Set the system mode."""
+        payload = {
+            "unique_id": broker.tcs.systemId,
+            "service": call.service,
+            "data": call.data,
+        }
+        hass.helpers.dispatcher.async_dispatcher_send(DOMAIN, payload)
+
+    async def set_zone_override(call):
+        """Set the zone override (setpoint)."""
+        entity_id = call.data[ATTR_ENTITY_ID]
+
+        registry = await async_get_registry(hass)
+        registry_entry = registry.async_get(entity_id)
```

Comment:
FYI I am going to work on a better way for an integration to be able to access its entities.